windows.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2017 The Virtual Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:generate go run generator.go
+
+// +build windows
+
+package virtual
+
+import (
+	"fmt"
+	"os"
+	"syscall"
+	"unsafe"
+)
+
+func init() {
+	registerBuiltins(map[int]Opcode{
+		dict.SID("_InterlockedCompareExchange"): InterlockedCompareExchange,
+		dict.SID("_beginthreadex"):              _beginthreadex,
+		dict.SID("_endthreadex"):                _endthreadex,
+		dict.SID("_msize"):                      _msize,
+		dict.SID("GetCurrentThreadId"):          GetCurrentThreadId,
+		dict.SID("GetLastError"):                GetLastError,
+	})
+}
+
+// TODO: use some generic wchar_16 stuff?
+func GoUTF16String(s uintptr) string {
+	ptr := (*[1 << 20]uint16)(unsafe.Pointer(s))
+	return syscall.UTF16ToString(ptr[:])
+}
+
+// DWORD WINAPI GetLastError(void);
+func (c *cpu) GetLastError() {
+	writeI32(c.rp, c.tlsp.errno)
+	c.setErrno(0)
+}
+
+// DWORD WINAPI GetCurrentThreadId(void);
+func (c *cpu) GetCurrentThreadId() {
+	writeU32(c.rp, uint32(c.tlsp.threadID))
+}
+
+// LONG __cdecl InterlockedCompareExchange(_Inout_ LONG volatile *Destination,_In_ LONG Exchange,_In_ LONG Comparand);
+func (c *cpu) InterlockedCompareExchange() {
+	// TODO: memory barrier: https://msdn.microsoft.com/de-de/library/windows/desktop/ms683560(v=vs.85).aspx
+	sp, comparand := popI32(c.sp)
+	sp, exchange := popI32(sp)
+	dest := readPtr(sp)
+
+	if strace {
+		fmt.Fprintf(os.Stderr, "InterlockedCompareExchange(%#x, %#x, %#x)\n", comparand, exchange, dest)
+	}
+
+	// TODO: currently we don't seem to support multiple threads, so we don't need to ensure
```

Comment:
Since there're no atomic operations yet, I'll just case that uintptr to *int32 and use
the sync package to implement it (to maybe save someone annoying bug tracking
and then noticing a todo comment). Thanks for the hint though.